src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggerAwarePass.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Psr\Log\LoggerAwareInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Automatically add setLogger method call to any service that implements Psr\Log\LoggerAwareInterface.
+ *
+ * @see http://www.php-fig.org/psr/psr-3/
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class LoggerAwarePass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('logger') && !$container->hasAlias('logger')) {
+            return;
+        }
+
+        $reference = new Reference('logger');
+
+        foreach ($container->getDefinitions() as $definition) {
+            if (!class_exists($definition->getClass()) || !$definition->isAutoconfigured()) {
```

Comment:
no reason to skip autoconfigured definitions. If we want `autoconfigure` to leverage this, the compiler pass should only process definitions with a given tag, and that tag+LoggerAwareInterface should be registered for autoconfiguration (see https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L269).